common/lib/xmodule/xmodule/x_module.py

```diff
@@ -1613,10 +1613,10 @@ def get_course_template_context(self):
         request.user = user
         user_info = cc.User.from_django_user(self.runtime.user).to_dict()
         course_id = self.runtime.course_id
-        course = get_course_with_access(self.runtime.user, 'load_forum', course_id)
+        course = get_course_with_access(self.runtime.user, 'load', course_id, check_if_enrolled=True)
         user_cohort_id = get_cohort_id(user, course_id)
 
-        unsafethreads, query_params = get_threads(request, course_id)
+        unsafethreads, query_params = get_threads(request, course)
```

Comment:
Also, having to use a fake request object outside of test code is usually a bad sign :-/